testing/matrix/linter/rules/rules.go

```diff
@@ -339,6 +340,46 @@ func newConvertError(object storage.StoreObject, err error) errors.LintRuleError
 	)
 }
 
+func objectRevisionHistoryLimit(object storage.StoreObject) errors.LintRuleError {
+	if object.Unstructured.GetKind() == "Deployment" {
+		converter := runtime.DefaultUnstructuredConverter
+		deployment := new(appsv1.Deployment)
+
+		err := converter.FromUnstructured(object.Unstructured.UnstructuredContent(), deployment)
+		if err != nil {
+			return newConvertError(object, err)
+		}
+
+		// https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#revision-history-limit
+		// Revision history limit controls the number of replicasets stored in the cluster for each deployment.
+		// Higher number means higher resource consumption, lower means inability to rollback.
+		//
+		// Since Deckhouse does not use rollback, we can set it to 2 to be able to manually check the previous version.
+		// It is more important to reduce the control plane pressure.
+		idealHistoryLimit := int32(2)
```

Comment:
desiredHistoryLimit ?

Reply:
I picked this term because I had seen this word in the doc
>  By default, 10 old ReplicaSets will be kept, however its ideal value depends on the frequency and stability of new Deployments.

It is ok to change the name, but it is not a `desired` value. It is the maximum allowed value, thus I will use the term `max` there.